pkg/dcmtime/common.go

```diff
@@ -0,0 +1,147 @@
+package dcmtime
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// isIncluded returns whether `check` is included in `limit`.
+//
+// Example: to test whether seconds should be included, you would:
+// isIncluded(PrecisionSeconds, [caller-passed-limit])
+func isIncluded(check PrecisionLevel, precision PrecisionLevel) bool {
+	return check <= precision
+}
+
+// truncateMilliseconds truncate nanosecond time.Time value to arbitrary precision.
+func truncateMilliseconds(
+	nanoSeconds int, precision PrecisionLevel,
+) (millisecondsStr string) {
+	milliseconds := nanoSeconds / 1000
```

Comment:
minor nit: you could consider renaming `millis` to be shorter but similarly expressive if you felt like it